sklearn/neighbors/dist_metrics.pyx

```diff
@@ -1048,6 +1052,22 @@ cdef class HaversineDistance(DistanceMetric):
 #            norm2 += x2[j] * x2[j]
 #        return 1.0 - d / sqrt(norm1 * norm2)
 
+#------------------------------------------------------------
+# Arccos Distance
+#  D(x, y) = arccos(dot(x, y) / (|x| * |y|)) / PI
+# This is the metric version of cosine distance, i.e. the metric
+# most suitable when you wish to use cosine distance.
+
+cdef class ArccosDistance(DistanceMetric):
+    cdef inline DTYPE_t dist(self, DTYPE_t* x1, DTYPE_t* x2, ITYPE_t size) nogil except -1:
+        cdef DTYPE_t d = 0, norm1 = 0, norm2 = 0
+        cdef np.intp_t j
+        for j in range(size):
+            d += x1[j] * x2[j]
```

Comment:
That certainly true, but it does assume the data is normalized -- I was hoping to remain general.